ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -205,7 +207,7 @@ void performRecovery()
    */
   void pruneHardDeleteRecoveryRange() {
     StoreFindToken logFlushedTillToken = (StoreFindToken) startTokenSafeToPersist;
-    if (logFlushedTillToken != null && !logFlushedTillToken.isUninitialized()) {
+    if (logFlushedTillToken != null && !logFlushedTillToken.getType().equals(StoreFindToken.Type.Uninitialized)) {
```

Comment:
Because there is an equivalent check - the method wasn't adding value like it was before.